[BUG] test_window_running failed in 3.1.2+3.1.3

**Describe the bug**
related to https://github.com/NVIDIA/spark-rapids/pull/9489
We found this failure in 3.1.2 and 3.1.3 integration tests. (other versions passed cases correctly)


```
= 64 failed, 18948 passed, 2124 skipped, 680 xfailed, 279 xpassed, 864 warnings in 6377.49s (1:46:17) =
```

```
pyspark.sql.utils.ParseException: 
mismatched input 'NULLS' expecting {<EOF>, ';'}(line 1, pos 962)

== SQL ==
select row_number() over (order by a rows between UNBOUNDED PRECEDING AND CURRENT ROW) as row_num, rank() over (order by a rows between UNBOUNDED PRECEDING AND CURRENT ROW) as rank_val, dense_rank() over (order by a rows between UNBOUNDED PRECEDING AND CURRENT ROW) as dense_rank_val, count(b) over (order by a rows between UNBOUNDED PRECEDING AND CURRENT ROW) as count_col, min(b) over (order by a rows between UNBOUNDED PRECEDING AND CURRENT ROW) as min_col, max(b) over (order by a rows between UNBOUNDED PRECEDING AND CURRENT ROW) as max_col, FIRST(b) OVER (ORDER BY a ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW) AS first_keep_nulls, FIRST(b, TRUE) OVER (ORDER BY a ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW) AS first_ignore_nulls, NTH_VALUE(b, 1) OVER (ORDER BY a ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW) AS nth_1_keep_nulls, sum(b) over (order by a rows between UNBOUNDED PRECEDING AND CURRENT ROW) as sum_col, NTH_VALUE(b, 1) IGNORE NULLS OVER (ORDER BY a ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW) AS nth_1_ignore_nulls from window_agg_table 
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------^^^
Stacktrace
b_gen = Byte, batch_size = '1000'

    @pytest.mark.parametrize('batch_size', ['1000', '1g'], ids=idfn) # set the batch size so we can test multiple stream batches
    @pytest.mark.parametrize('b_gen', all_basic_gens + [decimal_gen_32bit, decimal_gen_128bit], ids=meta_idfn('data:'))
    def test_window_running_no_part(b_gen, batch_size):
        conf = {'spark.rapids.sql.batchSizeBytes': batch_size,
                'spark.rapids.sql.castFloatToDecimal.enabled': True}
        query_parts = ['row_number() over (order by a rows between UNBOUNDED PRECEDING AND CURRENT ROW) as row_num',
                'rank() over (order by a rows between UNBOUNDED PRECEDING AND CURRENT ROW) as rank_val',
                'dense_rank() over (order by a rows between UNBOUNDED PRECEDING AND CURRENT ROW) as dense_rank_val',
                'count(b) over (order by a rows between UNBOUNDED PRECEDING AND CURRENT ROW) as count_col',
                'min(b) over (order by a rows between UNBOUNDED PRECEDING AND CURRENT ROW) as min_col',
                'max(b) over (order by a rows between UNBOUNDED PRECEDING AND CURRENT ROW) as max_col',
                'FIRST(b) OVER (ORDER BY a ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW) AS first_keep_nulls',
                'FIRST(b, TRUE) OVER (ORDER BY a ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW) AS first_ignore_nulls',
                'NTH_VALUE(b, 1) OVER (ORDER BY a ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW) AS nth_1_keep_nulls']
        if isinstance(b_gen.data_type, NumericType) and not isinstance(b_gen, FloatGen) and not isinstance(b_gen, DoubleGen):
            query_parts.append('sum(b) over (order by a rows between UNBOUNDED PRECEDING AND CURRENT ROW) as sum_col')
    
        if spark_version() > "3.1.1":
            query_parts.append('NTH_VALUE(b, 1) IGNORE NULLS OVER '
                               '(ORDER BY a ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW) AS nth_1_ignore_nulls')
    
>       assert_gpu_and_cpu_are_equal_sql(
            lambda spark : two_col_df(spark, UniqueLongGen(), b_gen, length=1024 * 14),
            "window_agg_table",
            'select ' +
            ', '.join(query_parts) +
            ' from window_agg_table ',
            validate_execs_in_gpu_plan = ['GpuRunningWindowExec'],
            conf = conf)

../../src/main/python/window_function_test.py:508: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../src/main/python/asserts.py:623: in assert_gpu_and_cpu_are_equal_sql
    assert_gpu_and_cpu_are_equal_collect(do_it_all, conf, is_cpu_first=is_cpu_first)
../../src/main/python/asserts.py:581: in assert_gpu_and_cpu_are_equal_collect
    _assert_gpu_and_cpu_are_equal(func, 'COLLECT', conf=conf, is_cpu_first=is_cpu_first, result_canonicalize_func_before_compare=result_canonicalize_func_before_compare)
../../src/main/python/asserts.py:486: in _assert_gpu_and_cpu_are_equal
    from_cpu = run_on_cpu()
../../src/main/python/asserts.py:471: in run_on_cpu
    from_cpu = with_cpu_session(bring_back, conf=conf)
../../src/main/python/spark_session.py:116: in with_cpu_session
    return with_spark_session(func, conf=copy)
../../src/main/python/spark_session.py:100: in with_spark_session
    ret = func(_spark)
../../src/main/python/asserts.py:205: in <lambda>
    bring_back = lambda spark: limit_func(spark).collect()
../../src/main/python/asserts.py:622: in do_it_all
    return spark.sql(sql)
../../../spark-3.1.2-bin-hadoop3.2/python/pyspark/sql/session.py:723: in sql
    return DataFrame(self._jsparkSession.sql(sqlQuery), self._wrapped)
/home/jenkins/agent/workspace/jenkins-rapids_it-AQE-dev-github-421/jars/spark-3.1.2-bin-hadoop3.2/python/lib/py4j-0.10.9-src.zip/py4j/java_gateway.py:1304: in __call__
    return_value = get_return_value(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

a = ('xro1119519', <py4j.java_gateway.GatewayClient object at 0x7f36717ff880>, 'o74', 'sql')
kw = {}
converted = ParseException("\nmismatched input 'NULLS' expecting {<EOF>, ';'}(line 1, pos 962)\n\n== SQL ==\nselect row_number() o...:79)\n\tat py4j.GatewayConnection.run(GatewayConnection.java:238)\n\tat java.lang.Thread.run(Thread.java:750)\n", None)

    def deco(*a, **kw):
        try:
            return f(*a, **kw)
        except py4j.protocol.Py4JJavaError as e:
            converted = convert_exception(e.java_exception)
            if not isinstance(converted, UnknownException):
                # Hide where the exception came from that shows a non-Pythonic
                # JVM exception message.
>               raise converted from None
E               pyspark.sql.utils.ParseException: 
E               mismatched input 'NULLS' expecting {<EOF>, ';'}(line 1, pos 962)
E               
E               == SQL ==
E               select row_number() over (order by a rows between UNBOUNDED PRECEDING AND CURRENT ROW) as row_num, rank() over (order by a rows between UNBOUNDED PRECEDING AND CURRENT ROW) as rank_val, dense_rank() over (order by a rows between UNBOUNDED PRECEDING AND CURRENT ROW) as dense_rank_val, count(b) over (order by a rows between UNBOUNDED PRECEDING AND CURRENT ROW) as count_col, min(b) over (order by a rows between UNBOUNDED PRECEDING AND CURRENT ROW) as min_col, max(b) over (order by a rows between UNBOUNDED PRECEDING AND CURRENT ROW) as max_col, FIRST(b) OVER (ORDER BY a ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW) AS first_keep_nulls, FIRST(b, TRUE) OVER (ORDER BY a ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW) AS first_ignore_nulls, NTH_VALUE(b, 1) OVER (ORDER BY a ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW) AS nth_1_keep_nulls, sum(b) over (order by a rows between UNBOUNDED PRECEDING AND CURRENT ROW) as sum_col, NTH_VALUE(b, 1) IGNORE NULLS OVER (ORDER BY a ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW) AS nth_1_ignore_nulls from window_agg_table 
E               --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------^^^

../../../spark-3.1.2-bin-hadoop3.2/python/pyspark/sql/utils.py:117: ParseException
```

**Steps/Code to reproduce bug**
run test case in spark 3.1.2 or 3.1.3

**Expected behavior**
Pass the case

**Environment details (please complete the following information)**
 - Environment location: [Standalone, YARN, Kubernetes, Cloud(specify cloud provider)]
 - Spark configuration settings related to the issue

**Additional context**
Add any other context about the problem here.

--- a/integration_tests/src/main/python/window_function_test.py
+++ b/integration_tests/src/main/python/window_function_test.py
@@ -501,7 +501,8 @@ def test_window_running_no_part(b_gen, batch_size):
     if isinstance(b_gen.data_type, NumericType) and not isinstance(b_gen, FloatGen) and not isinstance(b_gen, DoubleGen):
         query_parts.append('sum(b) over (order by a rows between UNBOUNDED PRECEDING AND CURRENT ROW) as sum_col')
 
-    if spark_version() > "3.1.1":
+    # The option to IGNORE NULLS in NTH_VALUE is not available prior to Spark 3.2.1.
+    if spark_version() >= "3.2.1":
         query_parts.append('NTH_VALUE(b, 1) IGNORE NULLS OVER '
                            '(ORDER BY a ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW) AS nth_1_ignore_nulls')
 
@@ -620,8 +621,8 @@ def test_window_running(b_gen, c_gen, batch_size):
     if isinstance(c_gen.data_type, NumericType) and (not isinstance(c_gen, FloatGen)) and (not isinstance(c_gen, DoubleGen)) and (not isinstance(c_gen, DecimalGen)):
         query_parts.append('sum(c) over (partition by b order by a rows between UNBOUNDED PRECEDING AND CURRENT ROW) as sum_col')
 
-    # The option to IGNORE NULLS in NTH_VALUE is not available prior to Spark 3.1.2.
-    if spark_version() > "3.1.1":
+    # The option to IGNORE NULLS in NTH_VALUE is not available prior to Spark 3.2.1.
+    if spark_version() >= "3.2.1":
         query_parts.append('NTH_VALUE(c, 1) IGNORE NULLS OVER '
                            '(PARTITION BY b ORDER BY a ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW) AS nth_1_ignore_nulls')
 

